src/main/java/seedu/address/model/partnercourse/PartnerUnit.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.model.partnercourse;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Partner Course's unit in SEPlendid.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPartnerUnit(Object)}
+ */
+public class PartnerUnit {
+
+    // Used in AppUtil#checkArgument and ParserUtil, JsonAdaptedXXX exceptions
+    public static final String MESSAGE_CONSTRAINTS =
+        "PartnerUnit can only take non-negative numeric values and should not be blank";
+
+    public final Double value;
+
+    /**
+     * Constructs an {@code PartnerUnit}.
+     *
+     * @param partnerUnit A valid partner unit.
+     */
+    public PartnerUnit(String partnerUnit) {
+        requireNonNull(partnerUnit);
+        checkArgument(isValidPartnerUnit(partnerUnit), MESSAGE_CONSTRAINTS);
+        value = Double.parseDouble(partnerUnit);
+    }
+
+    /**
+     * Constructs an {@code PartnerUnit}.
+     *
+     * @param partnerUnit A valid partner unit.
+     */
+    public PartnerUnit(Double partnerUnit) {
+        requireNonNull(partnerUnit);
+        checkArgument(isValidPartnerUnit(partnerUnit), MESSAGE_CONSTRAINTS);
+        value = partnerUnit;
+    }
+
+
+    /**
+     * Checks if the provided object is a valid partner unit.
+     * A valid partner unit is a positive number.
+     *
+     * @param test The object to be validated.
+     * @return true if the object is a valid partner unit, false otherwise.
+     */
+    public static boolean isValidPartnerUnit(Object test) {
+        if (test == null) {
+            // Handle null input gracefully
+            return false;
+        }
+
+        if (test instanceof String) {
+            // If the input is a String, attempt to parse it as a double
+            try {
+                double parsedValue = Double.parseDouble((String) test);
+
+                // Check if the parsed value is greater than 0
+                return parsedValue >= 0;
```

Comment:
Actually may I ask why we are accepting 0.0 as valid Local/PartnerUnit?

Reply:
0 mc mods exist :P